rotary_utils/repair.py

```diff
@@ -599,6 +433,44 @@ def stitch_all_contigs(circular_contig_tmp_fasta: str, bam_filepath: str, linkin
     return failed_contig_names
 
 
+class RepairPaths:
```

Comment:
@jmtsuji Here, I generated a series of data classes that store information that is normally stored in a large number of variables. These are similar to structs in C (https://en.wikipedia.org/wiki/Struct_(C_programming_language)). This allows you to pass a group of data to a function rather than a large number of variables.

When refactoring, you should look for clumps of data that can be passed together and create objects to represent them. This makes it easier to refactor functions.

Reply:
@LeeBergstrand This is a nice change -- the large number of inputs required for some of the functions was quite cumbersome. Right now, these classes are defined in the middle of the file, surrounded by other (mostly procedural) code. Is this normal, or should they be moved to the top of the file or to a different file, for clean organization? Thanks.

Reply:
> @LeeBergstrand This is a nice change -- the large number of inputs required for some of the functions was quite cumbersome. Right now, these classes are defined in the middle of the file, surrounded by other (mostly procedural) code. Is this normal, or should they be moved to the top of the file or to a different file, for clean organization? Thanks.
> 
> Update: I moved all class definitions to near the top of the code, just under `main`, for clarity.

If you can find a logical grouping to put them all in the same file, I would move them to separate files. For example, if there are classes and functions related to assembly, I would put them in `assembly.py.` `Repair.py` is getting long, so moving some of its supporting functions to other files may be reasonable.

Reply:
Thanks for the FYI - address this at least partially (as discussed below) in the latest round of refactoring for your review.